src/docs/arch-dhandle.dox

```diff
@@ -1,17 +1,85 @@
-/*! @arch_page arch-dhandle Data Handles and Btrees
+/*! @arch_page arch-dhandle Data Handles
 
-An internal structure called a Data Handle (dhandle) is used to represent and
-access Btrees and other data sources in WiredTiger. A dhandle is created
-when a table is accessed for the first time. It is kept in a global list
-and is shared across the sessions.  When a dhandle is not needed anymore
-and has been idle for a while, it is closed and destroyed, releasing all the
-resources associated with it.
+The data handle (dhandle) (\c WT_DATA_HANDLE) is a generic representation of
+any named data source. Dhandles are required to access any data source in the
+system. For the dhandle of B-Tree type, the \c void *handle field can be used
+like so \c (WT_BTREE *)dhandle->handle to get the B-Tree. For other data sources
```

Comment:
Normally \c only acts on the next word.  Can you please check the output to make sure that `\c void *handle` and `\c (WT_BTREE *)dhandle->handle` are doing the right thing?  If not, I think you can surround the text with "backticks": `